tensorflow/lite/kernels/unpack_test.cc

```diff
@@ -67,192 +66,72 @@ class UnpackOpModel : public SingleOpModel {
   std::vector<int> outputs_;
 };
 
-// float32 tests.
-TEST(UnpackOpTest, FloatThreeOutputs) {
-  UnpackOpModel<float> model({TensorType_FLOAT32, {3, 2}}, 0);
-  model.SetInput({1, 2, 3, 4, 5, 6});
-  model.Invoke();
+template <typename T>
+void Check(std::initializer_list<int> input_shape, int axis,
+           std::initializer_list<T> input_data,
+           std::vector<std::vector<int>> exp_output_shape,
+           std::vector<std::vector<T>> exp_output_data,
+           TensorType type = TensorType_FLOAT32) {
+  UnpackOpModel<T> m({type, input_shape}, axis);
+  m.SetInput(input_data);
+  m.Invoke();
 
   // Check outputs shapes.
-  const std::vector<std::vector<int>>& output_shapes = model.GetOutputShapes();
-  EXPECT_EQ(output_shapes.size(), 3);
-  EXPECT_THAT(output_shapes[0], ElementsAre(2));
-  EXPECT_THAT(output_shapes[1], ElementsAre(2));
-  EXPECT_THAT(output_shapes[2], ElementsAre(2));
+  EXPECT_THAT(m.GetOutputShapes(), ElementsAreArray(exp_output_shape));
 
   // Check outputs values.
-  const std::vector<std::vector<float>>& output_datas = model.GetOutputDatas();
-  EXPECT_EQ(output_datas.size(), 3);
-  EXPECT_THAT(output_datas[0], ElementsAre(1, 2));
-  EXPECT_THAT(output_datas[1], ElementsAre(3, 4));
-  EXPECT_THAT(output_datas[2], ElementsAre(5, 6));
+  EXPECT_THAT(m.GetOutputDatas(), ElementsAreArray(exp_output_data));
 }
 
-TEST(UnpackOpTest, FloatThreeOutputsAxisOne) {
-  UnpackOpModel<float> model({TensorType_FLOAT32, {3, 2}}, 1);
-  model.SetInput({1, 2, 3, 4, 5, 6});
-  model.Invoke();
-
-  // Check outputs shapes.
-  const std::vector<std::vector<int>>& output_shapes = model.GetOutputShapes();
-  EXPECT_EQ(output_shapes.size(), 2);
-  EXPECT_THAT(output_shapes[0], ElementsAre(3));
-  EXPECT_THAT(output_shapes[1], ElementsAre(3));
+// float32 tests.
+TEST(UnpackOpTest, FloatThreeOutputs) {
+  Check<float>({3, 2}, 0, {1, 2, 3, 4, 5, 6}, {{2}, {2}, {2}},
```

Comment:
can you add args documentation? thanks

Reply:
@renjie-liu 

added args documentation, please check once again, TIA.